ambry-store/src/main/java/com/github/ambry/store/BootstrapController.java

```diff
@@ -0,0 +1,203 @@
+            } else {
+              // R.complete -> R or FC.complete -> R
+              listenerToInvoke = storageManagerListener;
+              logStateChange("R.complete -> R or FC.complete -> R", partitionName);
+            }
+          }
+        }
+      }
+
+      assert listenerToInvoke != null;
+      listenerToInvoke.onPartitionBecomeBootstrapFromOffline(partitionName);
+      if (listenerToInvoke == fileCopyManagerListener) {
+        try {
+          storeManager.getPrimaryClusterParticipant().getReplicaSyncUpManager().waitForFileCopyCompleted(partitionName);
+        } catch (InterruptedException e) {
+          String message = "Failed `waitForFileCopyCompleted` step for " + partitionName;
+          logger.error(message);
+          throw new StateTransitionException(message, BootstrapControllerFailure);
+        }
+      }
+    }
+
+    private void logStateChange(String stateChange, String partitionName) {
+      logger.info("BootstrapController State change `{}` for partition `{}`", stateChange, partitionName);
+    }
+
+    @Override
+    public void onPartitionBecomeStandbyFromBootstrap(String partitionName) {
+      // no op
+    }
+
+    @Override
+    public void onPartitionBecomeLeaderFromStandby(String partitionName) {
+      // no op
+    }
+
+    @Override
+    public void onPartitionBecomeStandbyFromLeader(String partitionName) {
+      // no op
+    }
+
+    @Override
+    public void onPartitionBecomeInactiveFromStandby(String partitionName) {
+      // no op
+    }
+
+    @Override
+    public void onPartitionBecomeOfflineFromInactive(String partitionName) {
+      // no op
+    }
+
+    @Override
+    public void onPartitionBecomeDroppedFromOffline(String partitionName) {
+      // no op
+    }
+
+    private boolean isFileCopyFeatureEnabled() {
+      return storeConfig.fileCopyFeatureEnabled.equals("true");
+    }
+
+    private boolean isFileExists(
+        @Nonnull PartitionId partitionId, @Nonnull String fileName) {
+      return storeManager.isFileExists(partitionId, fileName);
+    }
+
+    private boolean isAnyLogSegmentExists(@Nonnull PartitionId partitionId) {
+      try {
+        return storeManager.isFilesExistForPattern(partitionId, allLogSegmentFilesPattern);
+      } catch (IOException e) {
+        String message = "Failed `isFilesExistForPattern` step for " + partitionId;
+        logger.error(message);
+        throw new StateTransitionException(message, BootstrapControllerFailure);
+      }
+    }
+
+    private void deleteFileCopyData(@Nonnull PartitionId partitionId) throws IOException, StoreException {
+      // Currently we’ll delete all datasets by removing this partition's BlobStore
+      // An optimisation could be explored to only delete incomplete datasets
+      // More about this can be found in this comment :-
+      // https://docs.google.com/document/d/1u57C0BU8oMMuMHC6Fh_B-6R612EaQb2AxtnBDlN6sAs/edit?disco=AAABZyj2rHo
+      storeManager.removeBlobStore(partitionId);
```

Comment:
We cannot delete the directory in this case.

Reply:
This is `FC.Incomplete -> R` case as we had discussed in this doc :- https://docs.google.com/document/d/1u57C0BU8oMMuMHC6Fh_B-6R612EaQb2AxtnBDlN6sAs/edit?tab=t.0#heading=h.seiobqi3qdu3

If we don't remove the BlobStore, should we then instead explicitly delete FC_files?

Reply:
Discussed with @aga9900 
We will need to evaluate the impact of removeBlobStore call here.
Added a TODO.